runner/src/builtins/llm.ts

```diff
@@ -75,52 +79,25 @@ export function llm(
   return (log: ReactivityLog) => {
     const thisRun = ++currentRun;
 
-    const { system, messages, prompt, stop, max_tokens, model } =
+    const { system, messages, prompt, stop, maxTokens, model } =
       inputsCell.getAsQueryResult([], log) ?? {};
 
-    // Define types for our parameters
-    type BaseParams = {
-      model: string;
-      messages: SimpleContent[] | SimpleMessage[];
-      max_tokens: number;
-      metadata?: Record<string, string>;
-    };
-
-    type StandardParams = BaseParams & {
-      system: string;
-      prompt: string;
-      stop: string;
-    };
-
-    type O1Params = BaseParams;
-
-    let llmParams: StandardParams | O1Params = {
+    // Unclear how `prompt` played a part here -- if and only if `openai:o1` used,
+    // and if `system` and `prompt` both defined (with lacking `messages`), that
+    // would then be used. Any special handling of models should be handled by toolshed.
+    //
+    // As of now, expecting `messages` to be populated, but `prompt` is currently unhandled.
+    const llmParams: LLMRequest = {
       system: system ?? "",
       messages: messages ?? [],
-      prompt: prompt ?? "",
       stop: stop ?? "",
-      max_tokens: max_tokens ?? 4096,
+      maxTokens: maxTokens ?? 4096,
       model: model ?? "claude-3-5-sonnet",
       metadata: {
         // FIXME(ja): how do we get the context of space/charm id here
         context: "charm",
       },
-    } as StandardParams;
-
-    // FIXME(ja): look at if model supports system messages instead...
-    // or perhaps we do this in the toolshed handler instead?
-    if (model?.startsWith("openai:o1")) {
-      const combinedMessage = system && prompt
-        ? (`${system}\n\n${prompt}` as SimpleContent)
-        : ((system || prompt) as SimpleContent);
-
-      llmParams = {
-        messages: messages ?? (combinedMessage ? [combinedMessage] : []),
-        model: model,
-        max_tokens: max_tokens ?? 4096,
-        metadata: llmParams.metadata,
-      };
-    }
```

Comment:
This could use another set of eyes. I'm not sure how this was supposed to work, with `prompt` never used AFAICT